modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache

```diff
@@ -10,6 +10,9 @@ import 'rxjs/add/operator/map';
 import * as models                                           from '../model/models';
 import { BASE_PATH, COLLECTION_FORMATS }                     from '../variables';
 import { Configuration }                                     from '../configuration';
+{{#withInterfaces}}
+import { I{{classname}} }                                    from './{{classname}}Interface';
```

Comment:
Hi @TiFu,

Sure we can rename both the interface and the class name, if @Vrolijkx and everyone else is happy with that. But this will virtually change the current class names and their file names, so if someone has imported and using `UserApi` resource _class_ at the moment, this implementation will replace it with the new interface, therefore will break their code.